Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -3021,6 +3021,28 @@ def visit_Node(self, node):
             node.in_nogil_context = True
         return node
 
+    def visit_ModuleNode(self, node):
+        # Not strictly a GIL check however check to see if numpy import_array is called
+        # should probably be done as late as possible
+        for mod in node.scope.cimported_modules:
+            if mod.name == "numpy":
+                import_array = mod.lookup_here("import_array")
+                _import_array = mod.lookup_here("_import_array")
+                if ((import_array and not import_array.used)
+                        or (_import_array and not _import_array.used)):
```

Comment:
Is `or` correct here? If either of the two _is_ used, we should not generate the call.

Reply:
Ah... `_import_array` is always used because it's used inline in the pxd file. I'll come back to this.